### PR TITLE
MMDP-322 Implement search and refractor's research section

### PR DESCRIPTION
**Pull request by Alameen688**
**What does this PR do?**

Implements search feature on the research section of the resources page

**Description of Task to be completed?**

- [x] An administrator should be able to search for a document displayed in the 'Research' section of the 'Resources' page by entering search criteria in the search box
- [x] Refractor research list implementation to use common components
- [x] Fix failing tests on users saga and component

**How should this be manually tested?**

- Checkout to this branch `ft-search-research-MMDP-332`
- RUN `yarn run:server` to start the backend server
- RUN `yarn client:dev` to start the frontend client
- Navigate to the research page, you should get a list of research, type in the title of the research you are looking for in the search box and click the search button. The research documents that match the search will appear in the list.

**Any background context you want to provide?**

 - none

**What are the relevant Jira issues?**

[MMDP-322](https://viisaus.atlassian.net/browse/MMDP-322)

**Screenshots (if appropriate)**

screen shot 2019-02-27 at 9 24 23 am
screen shot 2019-02-27 at 9 33 51 am

**Questions:**

### CODE REVIEW

**Comment by james-chege** [comment]
Good work on the search. You can consider working on the pagination, it seems that it is not work.

**Comment by james-chege** [comment]
When there is no results found for the search, the user doesn't get a message that there the search does not match anything in the records. You could consider adding the `No results` message

**Comment by Alameen688** [comment]
@james-chege Thanks for the catch. It's all fixed, those issues came up after rebasing, some things were modified that I didn't notice. The message has also been added.

**Comment by james-chege** [comment]
It also seems like the documents page has been broken by your rebase. You could also look into that.
screenshot 2019-02-25 at 04 49 29

**Comment by Alameen688** [comment]
@james-chege The design matches the mockup

File: client/__tests__/components/users/UsersDetails.test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* eslint-env jest */
 import React from 'react';
 import { mount } from 'enzyme';
-import UsersDetails from '../../../components/Users/USersDetails';
+import UsersDetails from '../../../components/Users/UsersDetails';
```
Comment (shemogumbe) [inline client/__tests__/components/users/UsersDetails.test.js]: Rename this file to UserDetails, in PascalCase naming we use singular names

File: client/containers/Users/ViewUsers.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Group from '../../components/common/GroupButtons';
-import UserViewList from '../../components/Users/USersDetails';
+import UserViewList from '../../components/Users/UsersDetails';
```
Comment (shemogumbe) [inline client/containers/Users/ViewUsers.js]: Same as above, rename to UserDetails

Comment (shemogumbe) [inline client/containers/Users/ViewUsers.js]: Also avoid using the term View within your file name, the file is already in views/ directory, you can simply call it UserList

Comment (Alameen688) [reply client/containers/Users/ViewUsers.js]: Those files were not created in this PR, they were first created in https://github.com/andela/mmdp_site/pull/22/files and the import name was renamed in this PR during the fix for case-sensitivity where Git was treating `UsersDetails` and `USersDetails` separately, therefore, causing issues during checkout (as complained by @mutaimwiti during today's standup) https://andela.slack.com/archives/CEQ5RJDC1/p1551171071006100

File: client/__tests__/components/users/UsersDetails.test.js (hunk 1 above)
Comment (Alameen688) [reply client/__tests__/components/users/UsersDetails.test.js]: Those files were not created in this PR, I only modified the import name due to the incorrect case. However, I will rename them all ASAP.

**Comment by james-chege** [comment]
> I have corrected the document page, it was caused due to the edit I made on the button style `button.ui.button` `width: 10em;` which I changed to `min-width: 10em` to fix the `Upload research` button text which takes two lines. That syle affects every button, I have modified the `btn__upload` style to fix just the `Upload Research` button.
> 
> @james-chege As for the design matching the mockup, this is not relating to this PR, the search section looks like it should. Another Jira issues relating to fixing all issues relating to those alignments should take care of that.
> I would have fixed it here but it requires refactoring the entire UI in the grid to use the common, as each card were given fixed width.
> 
> Apart from the suggestion to add the empty result text, most of the issues raised are existing issues on the section, and I have fixed the ones that can be fixed on this PR.

Alright @Alameen688 Can the pagination component position be fixed?

**Comment by Alameen688** [comment]
https://github.com/andela/mmdp_site/pull/57#issuecomment-467413769

@james I have decided to fix all issues on the research page with this PR.
I will re-request review once done.

**Comment by james-chege** [comment]
Good work @Alameen688. The implementation is fine.

**Pull request merged**